### ISSUE TITLE
Versions after v0.100.5 break mobile-toolkit apps built from angular-cli - dist/index.html output is undefined

### ISSUE THREAD

**Issue by Splaktar** [angular/universal#445]
* **I'm submitting a ...**
- [X ] bug report

* **What modules are related to this pull-request**
- [X] express-engine
- [ ] grunt-prerender
- [ ] gulp-prerender
- [ ] hapi-engine
- [X] preboot
- [ ] universal-preview
- [X] universal
- [ ] webpack-prerender

* **What is the current behavior?**

Unfortunately there is no error output at all from `ng build` on the angular-cli beta.6. This works fine when mobile-toolkit and universal are not part of the app. It also worked fine in angular-cli beta.5.

Contents of `dist/index.html`:
```html
undefined
```

After investigating, it's the `angular-universal` library change that caused the breakage. Reverting back from `0.101.5` to `0.100.5` fixes the failure. `0.101.9` and `0.102.1` also fail to output anything but `undefined` to `dist/index.html`.

* **What is the expected behavior?**

Contents of `dist/index.html` should be full of HTML.

* **Please tell us about your environment:**

- Angular version: 2.0.0-RC.1
- Browser: all
- Language: TypeScript 1.8.10
- OS:  Mac OS X El Capitan
- Platform: NodeJs

* **Other information**
https://github.com/angular/mobile-toolkit/issues/66

**Comment by MarkPieszak** [comment]
Hey Michael, I'll take a look and see what I can find. Are you using router-deprecated? It's all that's supported at the moment at least.

**Comment by Splaktar** [comment]
@MarkPieszak yes.

**Comment by Splaktar** [comment]
This is also being tracked here: https://github.com/angular/angular-cli/issues/958

**Comment by gdi2290** [comment]
@Splaktar universal has been updated to rc.3 via `0.103.2` can you check it out to see if it fixes the problem. Also set `async: true` in the universal config

**Comment by Splaktar** [comment]
@gdi2290 I will give it a shot after my talk tonight. Thank you!

**Comment by Splaktar** [comment]
This will take a bit more time to get to. 

For my actual app, I'm blocked by:
https://github.com/vaadin/angular2-polymer/issues/26

For a new app via `ng new testApp --mobile`, I'm blocked by:
https://github.com/angular/angular-cli/issues/1174

I might be able to disable angular2-polymer temporarily in order to test this in my app. Will try that when I get some more time.

**Comment by MarkPieszak** [comment]
Is the CLI using Universal 103.2? That one includes the new alpha router and those peerDep fixes.

**Comment by Splaktar** [comment]
@MarkPieszak angular-cli `beta.8` is still using angular-universal `0.100.5`.

I manually upgraded a new CLI project to `0.103.2` but then that threw errors because something apparently still depended on `@angular/router-deprecated`. So I added that, then it threw exceptions due to `The selector "app-root" did not match any elements`.

**Comment by Splaktar** [comment]
@gdi2290 Thank you for making a PR for the `async: true` setting: https://github.com/angular/angular-cli/pull/1163

**Comment by Splaktar** [comment]
I was able to verify this is working again with `0.103.3`: https://github.com/angular/angular-cli/issues/1174#issuecomment-228626726
